libwallet/assets/btc/sync.go

```diff
@@ -219,6 +219,10 @@ func (asset *Asset) handleNotifications() {
 
 notificationsLoop:
 	for {
+		if asset.syncCtx.Err() != nil {
+			break notificationsLoop // return early
+		}
```

Comment:
This isn't a good idea. Sync doesn't stop exactly when the ctx is canceled. Allow notifications to be processed until sync stops completely. Pages that no longer care about sync ntfns can de-register their progress listeners.

Reply:
Okay. We exit below on context cancellation and I think we should give it more priority? All pages must have been closed before this context is canceled.